### ISSUE TITLE
Add timeout for compiling and execution

### ISSUE THREAD

**Issue by stonemaster** [stonemaster/dlang-tour-rdmd#1]
The docker container should support timeout from inside (currently implemented from the outside) as another security feature.

> Example script:
> https://github.com/docker/docker/issues/1905

**Comment by stonemaster** [comment]
Example BASE64:

```
aW1wb3J0IHN0ZC5zdGRpbzsKCi8vIExldCdzIGdldCBnb2luZyEKdm9pZCBtYWluKCkgewogICAgd2hpbGUgKHRydWUpIHt3cml0ZWxuKCJIZWxsbyBXb3JsZCEgOy0pIik7IH0KfQ==
```

**Comment by stonemaster** [comment]
Fixes as per  8d0ddf7
